applications/matter_weather_station/src/main.cpp

```diff
@@ -47,7 +59,8 @@ int main()
 		goto exit;
 	}
 
-	err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
+	err = ConnectivityMgr().SetThreadDeviceType(
+		ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
```

Comment:
Somehow I missed it before. Why don't use the Sleepy End Device role? With Minimal End Device, Thingy:53 will constantly drain the battery on RX operation.

Reply:
@LuDuda I pushed changes adding SED config and necessary code in main.cpp: https://github.com/nrfconnect/sdk-nrf/pull/5265/files#diff-10f42255fd36402a402373f795896a1b82065e1f39ad2a61e9027df292b20716R62

Nevertheless I see few things for further consideration:
1. Should we really update poll period intervals manually in the application? I believe Matter Thread layer should take care of that based on the kconfigs values, but it would need some changes in the upstream.
2. I checked that CRI and CRA values are updated with set poll period values, but how the CHIPTool controller is going to behave, when it periodically asks for sensor data to refresh chart (I don't know how often but let's say 1 s) and response interval would be much bigger (e.g. 8 seconds)? That's why I set only 1 s poll period to not cause problems for know.
3. The controller should consider CRI and CRA values in regular communication. However during CASE session there is arbitrary timeout value set to 10 seconds. That would mean that we can't set bigger poll period than 10 seconds, or CASE will fail.